org.eclipse.dartboard/src/org/eclipse/dartboard/launch/LaunchShortcut.java

```diff
@@ -38,57 +44,80 @@
  * @author Jonas Hungershausen
  *
  */
+@SuppressWarnings("restriction")
 public class LaunchShortcut implements ILaunchShortcut {
 
 	@Override
 	public void launch(ISelection selection, String mode) {
+		// Get the active project from the
 		IProject selected = null;
 		if (selection instanceof StructuredSelection) {
 			Object firstElement = ((StructuredSelection) selection).getFirstElement();
-			if (firstElement instanceof IProject) {
-				selected = (IProject) firstElement;
+			if (firstElement instanceof IResource) {
+				selected = ((IResource) firstElement).getProject();
 			}
 		}
 
+		// Shouldn't happen
+		if (selected == null) {
+			MessageDialog.openError(null, Messages.Launch_ConfigurationRequired_Title,
```

Comment:
launchProject(IProject project, String mode)  should handle this dialog
 Just pass the project object and  mode

Reply:
Done.